From a9eaf1c98ed408fd6e8897fa9e2f99ed17db1ebc Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Tue, 23 Jul 2019 12:26:17 +0200
Subject: Port to KCalCore API changes

The code is in #if 0, but I tested compilation of this in the calligra
stable branch:
https://commits.kde.org/calligra/2d484fda1b31a72659088a4bfce5c3708e923cb0

So you can use KCalCore again if you want :-)
It's turning into a KF5 framework so the API will be stable from now on.

CCMAIL: danders@get2net.dk
---
 .../filters/icalendar/export/icalendarexport.cpp     | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/src/plugins/filters/icalendar/export/icalendarexport.cpp b/src/plugins/filters/icalendar/export/icalendarexport.cpp
index b21320f..828b641 100644
--- a/src/plugins/filters/icalendar/export/icalendarexport.cpp
+++ b/src/plugins/filters/icalendar/export/icalendarexport.cpp
@@ -35,6 +35,7 @@
 #include <kcalcore/attachment.h>
 #include <kcalcore/icalformat.h>
 #include <kcalcore/memorycalendar.h>
+#include <kcalcore_version.h>
 
 #include <QTextCodec>
 #include <QByteArray>
@@ -355,14 +356,15 @@ void ICalendarExport::createTodos(KCalCore::Calendar::Ptr cal, const Node *node,
         todo->setOrganizer(node->projectNode()->leader());
     }
     if ( node->type() != Node::Type_Project && ! node->leader().isEmpty()) {
-#if KCALCORE_HAVE_NO_PERSION_PTR
+#if KCALCORE_VERSION >= QT_VERSION_CHECK(5, 11, 80)
         KCalCore::Person p = KCalCore::Person::fromFullName(node->leader());
-        KCalCore::Attendee::Ptr a(new KCalCore::Attendee(p.name(), p.email()));
+        KCalCore::Attendee a(p.name(), p.email());
+        a.setRole(KCalCore::Attendee::NonParticipant);
 #else
         KCalCore::Person::Ptr p = KCalCore::Person::fromFullName(node->leader());
         KCalCore::Attendee::Ptr a(new KCalCore::Attendee(p->name(), p->email()));
-#endif
         a->setRole(KCalCore::Attendee::NonParticipant);
+#endif
         todo->addAttendee(a);
     }
     DateTime st = node->startTime(id);
@@ -381,13 +383,21 @@ void ICalendarExport::createTodos(KCalCore::Calendar::Ptr cal, const Node *node,
             const QList<Resource*> lst = task->requestedResources();
             foreach(const Resource *r, lst) {
                 if (r->type() == Resource::Type_Work) {
+#if KCALCORE_VERSION >= QT_VERSION_CHECK(5, 11, 80)
+                    todo->addAttendee(KCalCore::Attendee(r->name(), r->email()));
+#else
                     todo->addAttendee(KCalCore::Attendee::Ptr(new KCalCore::Attendee(r->name(), r->email())));
+#endif
                 }
             }
         } else {
             foreach(const Resource *r, s->resources()) {
                 if (r->type() == Resource::Type_Work) {
+#if KCALCORE_VERSION >= QT_VERSION_CHECK(5, 11, 80)
+                    todo->addAttendee(KCalCore::Attendee(r->name(), r->email()));
+#else
                     todo->addAttendee(KCalCore::Attendee::Ptr(new KCalCore::Attendee(r->name(), r->email())));
+#endif
                 }
             }
 
@@ -398,7 +408,11 @@ void ICalendarExport::createTodos(KCalCore::Calendar::Ptr cal, const Node *node,
         todo->setPercentComplete(task->completion().percentFinished());
     }
     foreach(const Document *doc, node->documents().documents()) {
+#if KCALCORE_VERSION >= QT_VERSION_CHECK(5, 11, 80)
+        todo->addAttachment(KCalCore::Attachment(doc->url().url()));
+#else
         todo->addAttachment(KCalCore::Attachment::Ptr(new KCalCore::Attachment(doc->url().url())));
+#endif
     }
     if (! parent.isNull()) {
         todo->setRelatedTo(parent->uid(), KCalCore::Incidence::RelTypeParent);
-- 
cgit v1.1
